internal/juju/applications.go

```diff
@@ -435,6 +541,28 @@ func (c applicationsClient) UpdateApplication(input *UpdateApplicationInput) err
 		}
 	}
 
+	// process expose
+	if input.Unexpose {
+		// TODO: (2022-08-10) right now we only unexpose all the possible endpoints.
+		// Additional logic must be set up to specify what endpoints
+		// should be unexpose in case this is simply a change, not a
+		// complete removal.
+		if err := applicationAPIClient.Unexpose(input.AppName, nil); err != nil {
+			log.Error().Err(err).Msg("error when trying to unexpose")
+			return err
+		}
+	} else if input.Expose != nil {
```

Comment:
We might also have a similar effect (with what you talk about in the TODO above) if we don't make the `input.Unexpose` and `input.Expose` mutually exclusive (i.e. `else if` -> `if`). Unexpose would close everything, then expose would open the listed endpoints. Though it should be only temporary, as it's two calls to the api rather than one.

Reply:
I agree. Additional logic has to be included to differentiate endpoints to be unexposed in following updates. At this point I will maintain this logic to have a first working version.